### ISSUE TITLE
When testing, result is different at different time with the same testing image.

### ISSUE THREAD

**Issue by MarkMoHR** [MarkMoHR/FastMaskRCNN_ForwardTest#2]
### Problem:
When testing, with the original Author's code, I meet a problem: The result is different(not color, but segmentation and recognition) at different time with the same testing image. The results of two different times are shown as below:
![Display](https://github.com/MarkMoHR/FastMaskRCNN_ForwardTest/raw/master/assets/diff1.png)
It is obviously that the result would be the same with the same trained model and testing image. So I try to locate the bug and the procedure is listed below:

### Bug locating:
1.  I found the bboxes are different at different testings, which would lead to the result of different masks, because a mask is predicted from the bbox.
2.  So I tested for 2 times and respectively printed the inter-mediate result of `outputs['rpn']['box']` / `outputs['roi']['box']` / `outputs['refined']['box']` / `outputs['final_boxes']['box']`. The result is that:
        **·**  `outputs['rpn']['box']` and `outputs['roi']['box']` are the **same**;
        **·**  `outputs['refined']['box']` and `outputs['final_boxes']['box']` are **different**. 
3. So it seems that the code  _**libs/nets/pyramid_network.py**_  [**Line300-313**](https://github.com/CharlesShang/FastMaskRCNN/blob/master/libs/nets/pyramid_network.py#L300-L313) have some problems: The weights of `slim.fully_connected` operations are always **randomly initialized**, not from the model.
4. Then I thought this might be caused by the `arg_scope`. The original arg_scope is from [Line211](https://github.com/CharlesShang/FastMaskRCNN/blob/master/libs/nets/pyramid_network.py#L211), the func `_extra_conv_arg_scope_with_bn` of which only defines the `slim.conv2d`, but no `slim.fully_connected`. I then found `slim.fully_connected` from another func `_extra_conv_arg_scope` at [Line64-79](https://github.com/CharlesShang/FastMaskRCNN/blob/master/libs/nets/pyramid_network.py#L64-L79). So I supposed that the authors neglected to add this arg_scope to the currently used func `_extra_conv_arg_scope_with_bn`

### Solution:
I added the arg_scope `slim.fully_connected` to `_extra_conv_arg_scope_with_bn` :
```
def _extra_conv_arg_scope_with_bn(weight_decay=0.00001,
                     activation_fn=None,
                     batch_norm_decay=0.997,
                     batch_norm_epsilon=1e-5,
                     batch_norm_scale=True):

  batch_norm_params = {
      'decay': batch_norm_decay,
      'epsilon': batch_norm_epsilon,
      'scale': batch_norm_scale,
      'updates_collections': tf.GraphKeys.UPDATE_OPS,
  }

  with slim.arg_scope(
      [slim.conv2d],
      weights_regularizer=slim.l2_regularizer(weight_decay),
      weights_initializer=slim.variance_scaling_initializer(),
      activation_fn=tf.nn.relu,
      normalizer_fn=slim.batch_norm,
      normalizer_params=batch_norm_params):
    with slim.arg_scope([slim.batch_norm], **batch_norm_params):
      with slim.arg_scope([slim.fully_connected],
          weights_regularizer=slim.l2_regularizer(weight_decay),
          weights_initializer=tf.truncated_normal_initializer(stddev=0.001),
          activation_fn=activation_fn,
          normalizer_fn=None):
        with slim.arg_scope([slim.max_pool2d], padding='SAME') as arg_sc:
            return arg_sc
```
and then tested for 2 times as well as printed the `outputs['final_boxes']['box']` / `outputs['mask']['mask']`. Finally I got **the same result** !

### Conclusion:
The author's original training code [libs/nets/pyramid_network.py](https://github.com/CharlesShang/FastMaskRCNN/blob/master/libs/nets/pyramid_network.py#L40-L62) have a bug in setting the arg_scope, which may cause some FC layers to **initialize their weights randomly**, not from the pre-trained model. So all the pre-trained model shared(trained with [the author's original code](https://github.com/CharlesShang/FastMaskRCNN)) online might have the same  problem mentioned above.

I suggested the authors should take this issue into account and try to fix this bug.
